### ISSUE TITLE
Missing DAG example of dynamic task mapping

### ISSUE THREAD

**Issue by Jorricks** [apache/airflow#26677]
### Description

While working on implementing [custom task notes](https://github.com/apache/airflow/pull/26457) I wanted to test my changes with [Dynamic Task mapping](https://airflow.apache.org/docs/apache-airflow/2.3.0/concepts/dynamic-task-mapping.html#simple-mapping), but we don't have any example DAGs for this.
We should add an example DAG containing this.

### Use case/motivation

For showing our users how this functionality can be used and also to improve testability for new features. 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by nikhil-koul** [comment]
Hi All,

I'm trying to implement Airflow expand functionality in DatabricksSubmitRunOperator.
If you, please provide me with examples of how it is done in DatabricksSubmitRunOperator?

Also, by using expand function, the tasks will be in parallel execution or sequential.

Regards,
Nikhil

**Comment by potiuk** [comment]
Please @nikhil-koul try to avoid repeating the same question in multiple places. I provide you guidelines on what you should do im https://github.com/apache/airflow/discussions/27050

**Comment by amoghrajesh** [comment]
@potiuk @Jorricks I see the PR was closed for this. What was the reason?
I have some bandwidth lying around, I can pick this issue up if someone gives me some context on what is expected and what does dynamic task mapping mean here?

**Comment by uranusjr** [comment]
In the PR thread the author metioned priorities so I assume it was simply lack of time. Feel free to raise a new PR based on those commits and apply the requested improvements (de-duplicating the example DAG and code in the documentation).

**Comment by amoghrajesh** [comment]
@Jorricks we need to close this issue. The PR pertaining to this got merged
